src/main.rs

```diff
@@ -221,7 +233,12 @@ fn notmain() -> Result<i32, anyhow::Error> {
         bail!("no probe was found")
     }
     log::debug!("found {} probes", probes.len());
-    let probe = probes[0].open()?;
+    let probe = if let Some(probe_opt) = opts.probe.as_deref() {
+        let selector: probe_rs::DebugProbeSelector = probe_opt.try_into()?;
+        probe_rs::Probe::open(selector)?
```

Comment:
ah, it seems that if you pass "$VID:$PID" as the selector and there are two connected probes with that VID PID pair then this will pick the "first one" of those two -- calling `probe-run` a second time results in a probe-rs error ("probe could not be created"). In that scenario we should throw an error and ask the user to narrow down the selection using the serial number.

Is it possible to use `DebugProbeSelector` to filter a list of probes?

Reply:
Right.

I don't think probe-rs provides a way to use a `DebugProbeSelector` to filter a list of probes. But it is easy enough to implement.

Updated version now requires that only 1 probe matches the `DebugProbeSelector` to continue.